src/ui/store/actions/editHistory.ts

```diff
@@ -0,0 +1,232 @@
+import {cloneDeep} from 'lodash';
+import {Dispatch} from 'redux';
+import {create as jdpCreate} from 'jsondiffpatch';
+
+import {DatalensGlobalState} from 'ui';
+
+import type {EditHistoryUnit, EditHistoryState, Diff} from '../reducers/editHistory';
+
+const PROPERTY_IGNORE_LIST = ['highchartsWidget'];
+
+const jdp = jdpCreate({
+    propertyFilter: function (name: string) {
+        return !PROPERTY_IGNORE_LIST.includes(name);
+    },
+});
+
+// Plugin for jsondiffpatch to diff functions by its' references
+const functionDiffFilter = (context: any) => {
```

Comment:
Is it really necessary to compare functions to navigate through the history?

Reply:
Yes, we have functions in store, for example in visualization.placeholders